Feedback

a-koz-0610 opened the pull request:
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `master` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `master` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `master`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](/help/leaving-feedback-in-github)”.
</details>

a-koz-0610 committed:
Setting up GitHub Classroom Feedback

Dean110 committed:
Adding unit tests to start iteration 1.

Dean110 commented on spec/js/ClickCounter.spec.js:
```diff
@@ -0,0 +1,51 @@
+            });
+        }
+    });
+    /*
+    > As a user, I want to be able to purchase _Auto Clickers_, so that I don't have to keep clicking a button all day.0
+    - Can retrieve a _Clicking Companion_ count.
+    - Can add to the _Clicking Companion_ count.
+    - Subtract the amount of the _Clicking Companion_ cost from your click count.
+    */
```
You can move the content of this comment into the description of the unit test to make a more 'living' test suite.

Dean110 replied:
Ok, updated in a future commit.

Dean110 replied:
We need to talk about this,

Dean110 commented on spec/js/ClickCounter.spec.js:
```diff
@@ -0,0 +1,51 @@
+    beforeEach(() => {
+        underTest = new ClickCounter();
+    });
+    describe("FEATURE : Have a way to count clicks, recording them as donuts.", () => {
+        for (let i = 0; i < 10; i++) {
+            it("Should record " + i + " click(s), increasing the donut count to " + i + ".", () => {
+                expectOneToOneClickToDonutCount(i, underTest);
+            });
+        }
+    });
```
Clever!  Good use of variables to reduce duplicate code!

Dean110 commented on spec/js/ClickCounter.spec.js:
```diff
@@ -0,0 +1,51 @@
+
+});
+
+function expectOneToOneClickToDonutCount(expectedClickCount, clickCounter) {
+    recordClicks(expectedClickCount, clickCounter);
+    expect(clickCounter.getDonutCount()).toBe(expectedClickCount);
+}
+
+function recordClicks(clickCount, clickCounter) {
+    for (let i = 0; i < clickCount; i++) {
+        clickCounter.recordClick();
+    }
+}
```
Consider using ES-6 'arrow' functions.

Dean110 commented on src/js/ClickCounter.js:
```diff
@@ -0,0 +1,19 @@
+class ClickCounter {
+    constructor() {
+        this._autoClickerCount = 0;
+        this._donutCount = 0;
+    }
+    getAutoClickerCount() {
+        return this._autoClickerCount;
+    }
+    getDonutCount() {
+        return this._donutCount;
+    }
+    purchaseAutoClicker() {
+        this._donutCount -= 100;
+        this._autoClickerCount++;
+    }
+    recordClick() {
+        this._donutCount++;
+    }
+}
```
Good looking class so far!

Dean110 reviewed (commented):
Good looking code so far!  I like the direction you are going with this project and the testing.  Unfortunately the project only meets 2 of our 16 requirements for this to be considered a _**Passing**_  grade.  The current status is 🔴 _**Incomplete**_ 🔴.  There is time to add to this project and bring it up to _**Passing**_, you have until Friday at 9:30 to complete a re-work of this project.  Please reach out to me and schedule a time to go over any areas that might confuse you.

Dean110 committed:
Adding Cost Increase to Clicking companion.

- Increase the cost of the second AutoClicker by 10%
- Increase the cost of each additional AutoClicker by 10% of the previous cost.

Dean110 commented on spec/js/ClickCounter.spec.js:
```diff
@@ -36,7 +30,26 @@ describe("ClickCounter is a class that counts clicks and turns them into donuts,
+        - Increase the cost of the second _Clicking Companion_ by an additional ten percent to 110.
+        - Increase the cost of every additional _Clicking Companion_ by an additional ten percent.
+        */
+        it("Given a donut count of 210, purchasing 2 clickers should decrease the donut count to 0.", () => {
+            recordClicks(210, underTest);
+            underTest.purchaseAutoClicker();
+            underTest.purchaseAutoClicker();
+            expect(underTest.getDonutCount()).toBe(0);
+        });
+        it("Given a donut count of 331, purchasing 3 clickers should decrease the donut count to 0.", () => {
+            recordClicks(331, underTest);
+            underTest.purchaseAutoClicker();
+            underTest.purchaseAutoClicker();
+            underTest.purchaseAutoClicker();
+            expect(underTest.getDonutCount()).toBe(0);
+        });
```
Consider removing the duplicate logic by refactoring it into a function on the global scope.

Dean110 reviewed (commented):
Look for opportunities to refactor when they present themselves.

Dean110 committed:
Checking for ISF

- Throws error if availble donut count isn't enough to cover donut cost of auto clickers.

Dean110 committed:
AutoClicks affect the donut count when they are recorded.

Dean110 committed:
Iteration 1 done.

Dean110 committed:
Complete ClickCounter.js

Dean110 committed:
Starting testing of ClickButton
